### PR TITLE
refactor EditProfileActivity layout

### PR DESCRIPTION
**Pull request by connyduck**
the last major usages of `RelativeLayout`

### CODE REVIEW

File: app/src/main/res/layout/activity_edit_profile.xml [hunk 1]
```diff
@@ -14,77 +14,82 @@
...
+                app:srcCompat="@drawable/ic_add_a_photo_32dp" />
+
+            <ProgressBar
+                android:id="@+id/headerProgressBar"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:indeterminate="true"
+                android:visibility="gone"
+                app:layout_constraintBottom_toBottomOf="@id/headerPreview"
+                app:layout_constraintEnd_toEndOf="parent"
```
Comment (charlag) [inline app/src/main/res/layout/activity_edit_profile.xml]: If you want to center it in the header, maybe say it explicitly and not to the parent?

File: app/src/main/res/layout/activity_edit_profile.xml [hunk 2]
```diff
@@ -14,77 +14,82 @@
...
+                android:background="@drawable/round_button"
+                android:contentDescription="@string/label_avatar"
+                android:elevation="4dp"
+                app:layout_constraintBottom_toBottomOf="@id/headerPreview"
+                app:layout_constraintStart_toStartOf="@id/contentContainer"
+                app:layout_constraintTop_toBottomOf="@id/headerPreview"
+                app:srcCompat="@drawable/ic_add_a_photo_32dp" />
+
+            <ProgressBar
+                android:id="@+id/avatarProgressBar"
```
Comment (charlag) [inline app/src/main/res/layout/activity_edit_profile.xml]: This one is mispositioned

File: app/src/main/res/layout/activity_edit_profile.xml [hunk 3]
```diff
@@ -14,77 +14,82 @@
...
+                android:layout_centerInParent="true"
+                android:layout_marginStart="16dp"
+                android:indeterminate="true"
+                android:visibility="gone"
+                app:layout_constraintBottom_toBottomOf="@id/headerPreview"
+                app:layout_constraintStart_toStartOf="@id/contentContainer"
+                app:layout_constraintTop_toBottomOf="@id/headerPreview" />
 
             <LinearLayout
+                android:id="@+id/contentContainer"
```
Comment (charlag) [inline app/src/main/res/layout/activity_edit_profile.xml]: Do we still need this one? We could put it into `ConstraintLayout` maybe?

**Comment by charlag** [comment]
Looks good, glad that we eliminate them finally.
There are some small quirks.

File: app/src/main/res/layout/activity_edit_profile.xml (hunk 3 above)
Comment (connyduck) [reply app/src/main/res/layout/activity_edit_profile.xml]: yes i could, but I think the code is way easier to read with the additional layout so I'd rather leave it

**Pull request merged**